linera-base/src/vm.rs

```diff
@@ -63,5 +63,34 @@ pub enum EvmQuery {
     /// A read-only query.
     Query(Vec<u8>),
     /// A request to schedule an operation that can mutate the application state.
-    Mutation(Vec<u8>),
+    Operation(Vec<u8>),
+}
+
+/// An EVM operation containing a value and argument data.
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct EvmOperation {
+    /// The amount being transferred.
+    pub value: alloy_primitives::U256,
+    /// The encoded argument data.
+    pub argument: Vec<u8>,
+}
+
+/// Creates an operation from value and argument data.
+pub fn get_evm_operation(
```

Comment:
We can do better than this.
```
use crate::data_types::Amount;

impl EvmOperation {
    fn new(amount: Amount, argument: Vec<u8>) -> Self {
        Self { ... }
    }
    
    fn to_bytes(&self) -> Result<Vec<u8>, bcs::Error> { ... }
}

Reply:
Longer explanation:
* Even if one might suspect that there is a reason why it is not called `EvmOperation::new`, the function `get_evm_operation` sounds like it should return a value of type `EvmOperation`.
* Top-level `pub fn` definitions should be rare.

Reply:
Ok, done.
But top-level `fn` functions ought to be ok for test code?